saleor/order/utils.py

```diff
@@ -308,11 +311,39 @@ def sum_order_totals(qs, currency_code):
 
 
 def get_valid_shipping_methods_for_order(order: Order):
+    if not order.is_shipping_required():
+        return None
+    if not order.shipping_address:
+        return None
     return ShippingMethod.objects.applicable_shipping_methods_for_instance(
         order, channel_id=order.channel_id, price=order.get_subtotal().gross
     )
 
 
+def get_discounted_lines(lines, voucher):
```

Comment:
Function with the same name is in `saleor/checkout/utils.py`. The difference I see is that the one in `checkout` module uses `CheckoutLineInfo` dataclass for lines. Do we need both of them?